src/NHibernate/Hql/Ast/ANTLR/Tree/JoinSubqueryFromElement.cs

```diff
@@ -57,7 +57,9 @@ private IType CreateDataType(SelectClause selectClause, out IEntityPersister ent
 
 		public SqlString RenderText(SqlString subQuery, ISessionFactoryImplementor sessionFactory)
 		{
-			return RenderText(sessionFactory).Replace("{query}", subQuery.ToString());
+			var array = RenderText(sessionFactory).Split("{query}");
+
+			return new SqlString(new object[] {array[0], subQuery, array[1]});
```

Comment:
I think more optimal would be doing `IndexOf` and `Substring`. Something like:
```C#
var index = renderText.IndexOfOrdinal("{query}");
return new SqlString(
	renderText.Substring(0, index),
	subQuery,
	renderText.Substring(index + 7));
```

Reply:
Suggestion applied.